parsons/mobilecommons/mobilecommons.py

```diff
@@ -0,0 +1,122 @@
+        page_limit = 1000 if limit > 1000 or limit is None else limit
+        params += f'limit={str(page_limit)}'
+
+        logger.info(f'Working on fetching first {page_limit} rows. This can take a long time.'
+                    f'Each 1000 rows can take between 30-60 seconds to fetch.')
+
+        response = self.client.request(endpoint + self.companyid_param, 'GET', params=params)
+
+        # If good response, compile data into final_table
+        if response.status_code == 200:
+            # Parse xml to nested dictionary and load to parsons table
+            response_dict = xmltodict.parse(response.text, attr_prefix='', cdata_key='',
+                                            dict_constructor=dict)
+            response_table = Table(response_dict['response'][endpoint][data_key])
+            # Unpack any specified elements
+            for col in elements_to_unpack:
+                response_table.unpack_dict(col)
+            # Append to final table
+            final_table.concat(response_table)
+            # Check to see if there are more pages and determine how many to retrieve
+            avail_pages = int(response_dict['response'][endpoint]['page_count'])
+            req_pages = math.ceil(limit/page_limit)
+            pages_to_get = avail_pages if avail_pages < req_pages else req_pages
+            # Go fetch other pages of data
+            i = 2
+            while i <= pages_to_get:
+                page_params = params + f'&page={str(i)}'
+                logger.info(f'Fetching rows {str(i*page_limit)} - {str((i+1)*page_limit)} '
+                            f'of {limit}')
+                response = self.client.request(endpoint + self.companyid_param, 'GET',
+                                               params=page_params)
+                response_dict = xmltodict.parse(response.text, attr_prefix='', cdata_key='',
+                                                dict_constructor=dict)
+                response_table = Table(response_dict['response'][endpoint][data_key])
+                final_table.concat(response_table)
+                i += 1
+        # If initial response is bad, raise error
+        else:
+            error = f'Response Code {str(response.status_code)}'
+            error_html = BeautifulSoup(response.text, features='html.parser')
+            error += '\n' + error_html.h4.next
+            error += '\n' + error_html.p.next
+            raise HTTPError(error)
+
+        return final_table
+
+    def get_broadcasts(self, start_time=None, end_time=None, status=None, campaign_id=None,
+                         limit=None):
+        """
+        Retrieve broadcast data
+
+        `Args:`
+            :param start_time:
+            :param end_time:
+            :param status:
+            :param campaign_id:
+            :param limit:
+            :return:
+        """
+        # Still working on compiling the params, but this function does work atm with no params specified
+        params = ''
```

Comment:
if you use a dict for params, this should be fairly straightforward. you can also use the utility to parse dates.

```python
from parsons.utilities.datetime import parse_date

DATE_FMT = '%Y-%m-%d'
formatted_date = lambda x: parse_date(x).strftime(DATE_FMT)

...

params = {
    'start_time': formatted_date(start_time),
    'end_time': formatted_date(end_time),
}
```